packages/js-drive/lib/masternodes/manageMasternodesIdentities.js

```diff
@@ -0,0 +1,205 @@
+          // Create an identity for operator (charged from operator)
+          const operatorIdentityId = hash(Buffer.concat(
+            Buffer.from(masternodeEntry.proRegTxHash, 'hex'),
+            Buffer.from(proRegTxInfo.state.pubKeyOperator, 'hex'),
+          ));
+
+          await createIdentity(
+            operatorIdentityId,
+            Buffer.from(proRegTxInfo.state.pubKeyOperator, 'hex'),
+            IdentityPublicKey.TYPES.BLS12_381,
+          );
+
+          // Create a document in rewards data contract with percentage
+          documentsToCreate.push(transactionalDpp.document.create(
+            contract,
+            Buffer.from(masternodeEntry.proRegTxHash, 'hex'),
+            'masternodeRewardShares',
+            {
+              payToId: operatorIdentityId,
+              percentage: proRegTxInfo.operatorReward,
+            },
+          ));
+        }
+      }),
+    );
+
+    // PubKeyOperator is changed
+    const masternodesWithNewPubKeyOperator = currentMNList
+      .filter((currentMnListEntry) => previousMNList
+        // eslint-disable-next-line max-len
+        .find((previousMnListEntry) => previousMnListEntry.proRegTxHash === currentMnListEntry.proRegTxHash && previousMnListEntry.pubKeyOperator !== currentMnListEntry.pubKeyOperator));
+
+    await Promise.all(
+      masternodesWithNewPubKeyOperator.map(async (masternodeEntry) => {
+        const rawTransaction = await stateRepository
+          .fetchTransaction(masternodeEntry.proRegTxHash);
+
+        const { extraPayload: proRegTxInfo } = new Transaction(rawTransaction.data);
+
+        if (proRegTxInfo.operatorReward > 0) {
+          const operatorIdentityId = hash(Buffer.concat(
+            Buffer.from(masternodeEntry.proRegTxHash, 'hex'),
+            Buffer.from(proRegTxInfo.state.pubKeyOperator, 'hex'),
+          ));
+
+          const operatorIdentity = await stateRepository.fetchIdentity(operatorIdentityId);
+
+          //  Create an identity for operator if there is no identity exist with the same ID
+          if (operatorIdentity === null) {
+            await createIdentity(
+              operatorIdentityId,
+              Buffer.from(proRegTxInfo.state.pubKeyOperator, 'hex'),
+              IdentityPublicKey.TYPES.BLS12_381,
+            );
+          }
+
+          // Delete document from rewards data contract with ID corresponding to the
+          // masternode identity (ownerId) and previous operator identity (payToId)
+
+          const previousOperatorData = previousMNList
+            // eslint-disable-next-line max-len
+            .find((previousMnListEntry) => previousMnListEntry.proRegTxHash === masternodeEntry.proRegTxHash);
+
+          const previousOperatorIdentityId = hash(Buffer.concat(
+            Buffer.from(previousOperatorData.proRegTxHash, 'hex'),
+            Buffer.from(proRegTxInfo.state.pubKeyOperator, 'hex'),
+          ));
+
+          documentsToDelete.push(transactionalDpp.document.create(
```

Comment:
You need to fetch document with ownerId  and payToId and then you can pass it for delete (ID must be the same)